### PR TITLE
fix: store raster nodata value in table metadata

### PR DESCRIPTION
**Pull request by volaya**
## Issue

Fixes #

https://app.shortcut.com/cartoteam/story/307121/handle-nodata-values-in-raster-loader

## Proposed Changes

This PR adds the nodata value of the raster layer to the table metadata

## Pull Request Checklist

- [ x] I have tested the changes locally
- [ x] I have added tests to cover my changes (if applicable)
- [ ] I have updated the documentation (if applicable)

### CODE REVIEW

File: raster_loader/io.py [hunk 1]
```diff
@@ -585,6 +589,7 @@ def rasterio_windows_to_records(
             if output_quadbin:
                 rec = array_to_quadbin_record(
                     raster_dataset.read(band, window=window),
+                    raster_dataset.nodata,
```
[the remark this answers is not in the document]
Comment (volaya) [reply raster_loader/io.py:592]: If there aren't any nodata pixels, the nodata value can be defined anyway (in some file formats is mandatory to define it, in other it is not)

If it is not defined in the file, rasterio returns null, so the metadata would have null. We can decide if, in that case, we want to keep it like that, or maybe not add the "nodata" property to the metadata object

[the remark this answers is not in the document]
Comment (vdelacruzb) [reply raster_loader/io.py:592]: that is what i was wondering if there are situations where we don't introduce the nodata. Then we should at least take care of it when appending tables. In the function `write_metadata` you will see that there are two cases, append_records or not, for now maybe you are testing by using `--overwrite` so the process is never appending data

[the remark this answers is not in the document]
Comment (vdelacruzb) [reply raster_loader/io.py:592]: This topic as been discussed and `nodata` is introduced by row. So this implementation makes sense

**Review by vdelacruzb — APPROVED** [review]
LGTM

**Pull request merged**